src/Curiosity/Server.hs

```diff
@@ -674,7 +675,11 @@ run conf@Parse.ServerConf {..} runtime = liftIO $ do
   jwk <- SAuth.generateKey
   -- FIXME: See if this can be customized via parsing.
   let jwtSettings = SAuth.defaultJWTSettings jwk
-  Warp.run _serverPort $ waiApp jwtSettings
+  let warpSettings =
+        Warp.setPort _serverPort $
+        Warp.setBeforeMainLoop (SD.notifyReady >> pure ()) $
```

Comment:
Is it ok to have this bit of code always executed ? Shouldn't this be enabled behind some command-line flag (called from the systemd unit, and not otherwise) ?

Reply:
I usually don't guard that under a flag.

The `notifyReady` algorithm look like this:

1. looks for a `NOTIFY_SOCKET` env variable.
1. retrieve the notify socket from its value.
1. bind it.
1. send a UDP packet to it.

If it fails at any point, the function returns `Nothing` and stops its execution.

Outside systemd, it'll likely fail during the `NOTIFY_SOCKET` env variable lookup. Meaning, on runtime, the sole overhead will be this failed environment lookup. I wouldn't bother optimizing that away using a config flag.

Reply:
Thanks, it makes sense.